src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -75,6 +94,35 @@ namespace DB
         }
     }
 
+    template <typename NumericType, typename VectorType = ColumnVector<NumericType>>
+    static void fillColumnWithArrayNumericData(std::shared_ptr<arrow::ChunkedArray> & arrow_column, MutableColumnPtr & internal_column)
```

Comment:
Maybe we can combine this code with fillColumnWithNumericData to reduce code duplication? ListArray has methods like values() and offsets(), so we can firstly get offsets and then insert values without iterating over arrays.